### PR TITLE
Added documentation for rebalance auto-approval

### PR DESCRIPTION
**Pull request by ppatierno**
### Type of change

- Documentation

### Description

Documentation PR related to the #7310 which adds optimization proposal auto-approval.

### Checklist

- [x] Update documentation

### CODE REVIEW

File: documentation/modules/cruise-control/con-optimization-proposals.adoc [hunk 1]
```diff
@@ -274,3 +274,30 @@ Consider a shorter interval for fast changing clusters, although this increases
 * xref:con-optimization-goals-{context}[]
 * xref:proc-generating-optimization-proposals-{context}[]
 * xref:proc-approving-optimization-proposal-{context}[]
+
+== Auto-approving an optimization proposal
+
+In some cases, the manual approval process of the optimization proposal is not necessary.
```
Comment (scholzj) [inline documentation/modules/cruise-control/con-optimization-proposals.adoc:280]: Thsi sounds weird without listing the cases where it is not necessary. I would maybe phrase it as _if you want, you can have it automatically approved_.

**Review by scholzj — APPROVED** [review]

File: documentation/modules/cruise-control/con-optimization-proposals.adoc [hunk 2]
```diff
@@ -274,3 +274,30 @@ Consider a shorter interval for fast changing clusters, although this increases
...
+  labels:
+    strimzi.io/cluster: my-cluster
+  annotations:
+    strimzi.io/rebalance-auto-approval: true
+spec:
+  mode: # any mode
+  # ...
+----
+
+You are still able to see the optimization proposal summary in the `KafkaRebalance` status to be aware of the in progress rebalancing outcome.
```
[the remark this answers is not in the document]
Comment (ppatierno) [reply documentation/modules/cruise-control/con-optimization-proposals.adoc:303]: > What happens when the status is NotReady? Do we need to say anything about that or if an error occurs with auto approval?

Nothing to add, it's just normal flow as the manual approval.

> Maybe we don't need a procedure for automatically approving an optimization proposal, but maybe we should we say a bit more about what happens with a proposal when this process is automated? The flow, the steps. You generate and if the status is ProposalReady the proposal is automatically approved and a rebalance is initiated...

Not sure what to say to be honest. It is just ... auto approved and it goes through the same flow as you apply the "approve" annotation in the manual way.
Any suggestions if you want to add more?

File: documentation/modules/cruise-control/con-optimization-proposals.adoc [hunk 3]
```diff
@@ -274,3 +274,30 @@ Consider a shorter interval for fast changing clusters, although this increases
 * xref:con-optimization-goals-{context}[]
 * xref:proc-generating-optimization-proposals-{context}[]
 * xref:proc-approving-optimization-proposal-{context}[]
+
+== Auto-approving an optimization proposal
```
[the remark this answers is not in the document]
Comment (ppatierno) [reply documentation/modules/cruise-control/con-optimization-proposals.adoc:278]: Agree.

File: documentation/modules/cruise-control/con-optimization-proposals.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/cruise-control/con-optimization-proposals.adoc]: Okay. But do we want to change the title of [Approving an optimization proposal](https://strimzi.io/docs/operators/in-development/configuring.html#proc-approving-optimization-proposal-str) to _Manually approving?

And would we want to show this as an optional step in  [Generating optimization proposals](https://strimzi.io/docs/operators/in-development/configuring.html#proc-generating-optimization-proposals-str)

[the remark this answers is not in the document]
Comment (ppatierno) [reply documentation/modules/cruise-control/con-optimization-proposals.adoc]: > Okay. But do we want to change the title of [Approving an optimization proposal](https://strimzi.io/docs/operators/in-development/configuring.html#proc-approving-optimization-proposal-str) to Manually approving?

I have already changed the title.

File: documentation/modules/cruise-control/proc-generating-optimization-proposals.adoc [hunk 4]
```diff
@@ -147,6 +147,25 @@ spec:
   skipHardGoalCheck: true
 ----
 
+. Optionally, to have the optimization proposal to be automatically approved when it is ready, add the `strimzi.io/rebalance-auto-approval: true` annotation
```
Comment (PaulRMellor) [inline documentation/modules/cruise-control/proc-generating-optimization-proposals.adoc:150]:
```suggestion
. (Optional) To approve the optimization proposal automatically, set the `strimzi.io/rebalance-auto-approval` annotation to `true`:
```

**Review by PaulRMellor — APPROVED** [review]
Looks good

**Pull request merged**